### ISSUE TITLE
[1.0.0 released - additional requests made] Social networks block

### ISSUE THREAD

**Issue by robiso** [robiso/wondercms#42]
### [Plugin request] Social network(s) block
- Enable users to enter their social network links.
 - Every time a social network link is filled, its displayed somewhere on the page (footer block or something similar).

What social networks should this plugin include?
-  Open for discussion or chosen by plugin author(s).

**Comment by Tangol** [comment]
I have a better idea. 
Why not use https://oembed.com/ that would allow us to easily embed all kinds of links.

**Comment by robiso** [comment]
Not a bad idea, thanks for sharing. I welcome all pull-requests regarding this feature and implementation.

**Comment by PiersMorgan** [comment]
Hi there. I have actually come up with an implementation for this and i have incorporated it directly into a theme. It includes the social bar which has been asked for, and you do not even have to fill in text boxes/forms to update the links as i have built it in so that you can simply edit an editable field and edit the href links. This was far easier to implement into the existing code and also a lot more lightweight.

**Comment by robiso** [comment]
Hi @PiersMorgan.

Sounds perfect, I can't wait to take it for test drive.

We have a kind of "system" for rewarding new and active developers:
- this system includes publicly thanking you on our special contributors page: https://wondercms.com/special-contributors
- one time donation **if possible/when someone donates any funds**. (Since we don't get that many donations, I sometimes do it out of my own pocket)
- when there's a new version of WonderCMS, you get an additional honorable mention on:
  - https://wondercms.com/whatsnew and
  - https://github.com/robiso/wondercms/releases

Mind sharing your first/last name + website link? Once you publish your work, we'll add that info to the mentioned link. Additionally, your plugin will probably be published on the plugins (https://wondercms.com/plugins) page.

Let me know if you need any help testing this out. Here's also a link on how to publish a plugin: https://github.com/robiso/wondercms-plugins#how-to-submit-a-plugin, but I have a feeling you already know this! :)

**Comment by PiersMorgan** [comment]
Sure!

**Comment by PiersMorgan** [comment]
Shall i post here?

**Comment by robiso** [comment]
Looks pretty neat! 👍 

That's the least I can do for good hearted developers! For now, you've been added on:
- https://www.wondercms.com/special-contributors (special contributors page)
- https://www.wondercms.com/latest (WonderCMS download page)

You will be additionally to other mentioned links (what's new page, GitHub release notes) once we hit a new release, and it's not an if you get a donation, it's a matter of when (it's usually a symbolic $5 to $20 donation to the developer), hopefully we'll be able to afford larger rewards in the future.

**Comment by PiersMorgan** [comment]
Thanks so much! I will be sure to work on other features and solve more problems in the road-map.
:)

**Comment by robiso** [comment]
Be sure to take your time, WonderCMS is a slow pace development project, don't feel rushed into doing everything (**although it's awesome**, we don't want someone to burn out on WonderCMS 😄).

Your effort and time is appreciated, a sincere thank you @PiersMorgan, from me and every WonderCMS user. 👍

**Comment by PiersMorgan** [comment]
Hi there, i am getting a PHP error:
`Notice: Undefined property: stdClass::$date in C:\xampp\htdocs\dashboard\Feather\themes\blogTheme\theme.php on line 72`

Any ideas? or should i attempt a fix myself.

**Comment by robiso** [comment]
Looks like I created an error and left it at that stage:
The error shows on line 72, but the real problem is line 49:
`if (empty(wCMS::get('pages', wCMS::$currentPage->date))) { // Check if date doesn't exist for current page` (this statement isn't getting executed which means no date is saved to the database) - line 49 and 72 can be removed, since it's a sloppy way or "creating" a fetching the post date.

If you need me to remove it let me know, if re-send the ZIP when I get home.

**Comment by PiersMorgan** [comment]
Lines removed, all done :P God i wish git had personal messaging.

**Comment by PiersMorgan** [comment]
I have just finished converting the social bar to a plugin, it is fully working and the code is neatly added into the page when the plugin is added and it functions perfectly well.
This feature can now be closed.

**Comment by robiso** [comment]
Can you please create a release? (it can be created here - https://github.com/PiersMorgan/wonder-socialbar/releases).

I'm trying to install this plugin via my settings -> plugin installer and I get "error opening ZIP file".
If you want I can "fork" it to the WonderCMS plugins repository (https://github.com/robiso/wondercms-plugins) and I can create a release there. With that in mind, I forgot to mention to add a simple text "version" file, which indicates what version your plugin is (I added this to the general guidelines, thanks for reminding me:).

So in short: I can create a release and re-zip with a version file it to make sure it's installable through the Settings panel -> themes & plugins installer.

**Comment by PiersMorgan** [comment]
https://github.com/PiersMorgan/wonder-socialbar/releases/tag/1.0.0
Done, version file added also.

**Comment by robiso** [comment]
I re-tried installing this through the WonderCMS Settings panel -> Themes & plugins -> Install plugin and couldn't do it successfully.

I then checked the zip and saw why, there seemed to be a zip file within the original zip 🍡.
Here's the correct structure below (you can use this zip for your release, and it will be installable through the Settings panel).

[wonder-socialbar-1.0.0.zip](https://github.com/robiso/wondercms/files/1700456/wonder-socialbar-1.0.0.zip)

To recap the structure in the ZIP file attached above:
[plugin-folder-name] -> This folder should be zipped and uploaded on GitHub as a release (as you did).
  plugin-file.php

**Now successfully tested this**, a question:
- do I have to edit the files manually through FTP for the correct social network URLs to show?
- or is there some other functionality to edit them (this would make more sense)?

Mind sharing your PayPal account?

**Comment by PiersMorgan** [comment]
Paypal: acc.alt3@gmail.com
And yes / no social network links can be manually edited from the plugin.php but in the future i will make it so they can be edited from a text box or something.

**Comment by robiso** [comment]
Re-added your ZIP file above (I attached the wrong one first).

This seems like an inconvenience since at no part of WonderCMS do they have to access the site through FTP, I'll add a small note alongside publishing this plugin.

Donation sent. :)

**Comment by PiersMorgan** [comment]
I will work on re implementing the editable fields i have an idea just now for the next release. And thank you very much!

**Comment by robiso** [comment]
No problem. Be sure to replace the ZIP file with the one posted below, to make it installable through the Settings panel:

[social_bar.zip](https://github.com/robiso/wondercms/files/1700557/social_bar.zip)

**Comment by robiso** [comment]
@PiersMorgan, I'd just like to check up if there are any news or changes on the social networks block?

**Comment by robiso** [comment]
@PiersMorgan, any news on this plugin?

**Comment by robiso** [comment]
@PiersMorgan, closing due to inactivity. Please feel free to re-open if you would like to continue with development of this plugin.
